### PR TITLE
STM32: Add OS and Struct support

### LINKED ISSUE #2169: STM32: enable struct
like `os` this one is easy to enable and will let us use libraries for testing i2c, spi, etc :)

### PR DESCRIPTION
**Pull request by hierophect**
This PR adds rudimentary OS and Struct support, to enable more libraries for use. Does not yet integrate TRNG, see #2168. Closes  #2169

### CODE REVIEW

**Comment by hierophect** [comment]
Testing binary @ladyada 
[firmware.bin 2.zip](https://github.com/adafruit/circuitpython/files/3637337/firmware.bin.2.zip)

File: ports/stm32f4/mpconfigport.mk [hunk 1]
```diff
@@ -21,6 +21,8 @@ CIRCUITPY_ANALOGIO = 1
 CIRCUITPY_MICROCONTROLLER = 1
 CIRCUITPY_BUSIO = 1
 CIRCUITPY_TIME = 1
+CIRCUITPY_BUSIO = 1
```
Comment (ladyada) [inline ports/stm32f4/mpconfigport.mk]: woops this is a typo, please fix :)

**Comment by ladyada** [comment]
tested `struct` with 
https://www.journaldev.com/17401/python-struct-pack-unpack
with success

```
import struct
var = struct.pack('hhl', 5, 10, 15)
print(var)
print(struct.unpack('hhl', var))
```

**Comment by hierophect** [comment]
New bin
[firmware.bin 3.zip](https://github.com/adafruit/circuitpython/files/3637522/firmware.bin.3.zip)

**Comment by ladyada** [comment]
yay can merge once CI passes

**Pull request merged**